chore(docs): New Docs and Website

marshallswain opened the pull request:
Adds `/docs` at the root of the project.  The new docs have been moved from VuePress to VitePress.  Development is lightning fast.  All changes have been made on top of the default Vitepress template.  This means we will automatically get all upstream updates to the Vitepress template.  Additionally,

- Colors have been customized to fit our new design.  The new template uses UnoCSS under the hood, FTW.
- Dark mode and Algolia search are in place.
- Custom `Tabs` and `Tab` components have been created for code examples.  `Tab` accepts a `global-id` prop to synchronize switching between related sets of tabs.  So when you change to TypeScript, all tabs with a TypeScript example will switch, as well.  Each tab will also remember its active tab between page reloads.  (Remember that fenced code blocks will only work if they are wrapped in line breaks, top and bottom.)

See current state of docs at [dove.feathersjs.com](https://dove.feathersjs.com)

Pending:

- [x] Create page placeholders in same structure as crow docs.
- [x] Copy content from Dove docs
- [x] Figure out why links with anchors have the anchor stripped. (template issue in vitepress).  
   - I'm marking this as complete because I think it will be fixed once we deploy.  It's likely only and issue in dev mode.
- [x] Add existing cookbook content into the Ecosystem menu.
- [x] Rename /readme links to / (index.md)
- [x] Make header static on the home page
- [x] Remove all badges except npm
- [x] Hook up the "Get Started" buttons
- [x] Fix footer links
- [x] Finish replacing tabs
- [x] Fix meta-related icons
- [x] Fix migration guide
- [ ] Fix Footer styles on mobile. (have to debug directly on iPhone, because emulators don't display the issue)
- [ ] Replace placeholder text on the home page
- [ ] Add/Edit new content for Dove
- [ ] Fix Algolia Search
- [x] Setup redirect from `dove.docs.feathersjs.com` to `feathersjs.com`.
- [ ] Finish Partners Section on home page
- [x] Fix tabs on http://localhost:3333/api/services.html
- [x] Update feathers-knex, bring into core
- [x] Standalone utility for transpiling the TS files
- [ ] Update the feathers-knex link, here: http://localhost:3333/guides/basics/services.html#core-database-adapters
- [ ] Adapters - Memory page
   - [x] Update badges
   - [ ] Update examples
- [ ] Adapters - MongoDB page
   - [x] Update badges
   - [ ] Update examples
- [ ] Adapters - Knex page
   - [x] Update badges
   - [ ] Update examples
- [ ] OAuth
- [ ] Illustrate concepts for where to use things:
   - [ ] Resolvers are for changes to data or populating based on properties
   - [ ] Hooks are for side effects.
   - [ ] Services are for anything that maps cleanly to CRUD operations.
- [x] Externalize code snippets: https://vitepress.vuejs.org/guide/markdown.html#import-code-snippets
- [ ] Guide
   - [ ] Redo guide based on new generator
   - [ ] Show off channels
   - [ ] Front End Framework
- [ ] Stackblitz
- [x] Add error pages
- [ ] Update migration guide with section on whitelisting a custom param using schemas.
- [ ] Notes/docs on working with schemas
   - [ ] Schemas should be object types, not arrays.  This is required for clean type conversion/inference.
   - [ ] If the inferred TypeScript type suddenly becomes `never`, check for errors in the schema.  The most likely error is having a string in the  `required` property that doesn't exist in the `properties` object.
- [ ] Allow `resolveAll` [data resolvers](https://github.com/feathersjs/feathers/blob/dove/packages/schema/src/hooks/resolve.ts#L42-L44) to be optional.  PR with tests.
- [ ] Document a troubleshooting step for when `app.service('my-service').hooks(myServiceHooks)` fails with a TS error due to an invalid return value when upgrading or registering `regular` hooks as `around` hooks.
- [ ] Make a color-coded hooks console.log example.
- [ ] When customizing a service, make sure to remove or add the method from the `methods` array during service registration.
- [ ] Troubleshooting when schema type is unknown. 
   - [ ] Required properties must match properties in the schema.
- [x] Add a Troubleshooting Guide under the Migration Guide in the More section of the guides.
- [ ] Talk about feathers-sync and real-time provider (currently Socket.io) scaling.  Nobody else has really done this.  Socket scaling is generally a pain in the ass.  We've got it covered.
- [ ] Talk about how channels performance is O^1 and how we refactored Feathers to allow this level of performance.

![Screen Shot 2022-06-13 at 5 31 09 PM](https://user-images.githubusercontent.com/128857/173464022-d31b9d99-c233-40fa-82fc-e61b0224c8e6.jpg)

marshallswain committed:
feat: new docs site

marshallswain committed:
move over docs pages

marshallswain committed:
fix: update docs with deep outline

marshallswain committed:
fix: clean up the main navbar’s styles

- home page navbar is now static.  Blur removed
- docs pages navbar blur is 2px instead of 6px, and a translucent bg matching the background color was added to soften the text a bit and make the toolbar text easier to read.

marshallswain committed:
fix: update tagline and slide hero up under navbar

marshallswain committed:
fix: deep outline for application page

marshallswain committed:
feat: Add LanguageSelect component to sidebar

marshallswain committed:
chore: format on save in vscode settings

marshallswain committed:
fix: don’t prefetch in build script

marshallswain committed:
fix: don’t sync global language on tab change

Now, only the LanguageSelect will change all tabs. This makes it possible to change tabs without changing ALL global-id-matching tabs.

marshallswain committed:
feat: Tabs component hides tabs by default

- added `showTabs` prop to explicitly opt in to showing the tabs.

marshallswain committed:
feat: show a label on the top-left corner of tabs

When `Tabs` don’t show clickable tabs, each nested `Tab` component will show a label in its top-left corner.

marshallswain committed:
fix: routes to readme changed to index, match old links

marshallswain committed:
chore: clean up top nav, lint the config

marshallswain committed:
update packages

marshallswain committed:
Add cookbook to the Ecosystem dropdown

marshallswain committed:
Add cookbook to faq

marshallswain commented:
![Screen Shot 2022-06-15 at 8 22 56 AM](https://user-images.githubusercontent.com/128857/173851241-b52de3b8-468d-4ac3-a4b3-eaae9c8b7f77.jpg)

marshallswain committed:
Fix all badges

marshallswain committed:
clean up titles and selective linting fixes

marshallswain committed:
Update tagline and readme image

marshallswain committed:
Fix CTA buttons and add comparisons section

marshallswain committed:
begin replacing tabs on some api pages

marshallswain committed:
oauth page tabs

daffl committed:
Start updating Quick Start guide

marshallswain committed:
Use simpler header img

marshallswain committed:
Replace remaining Tabs.  Put TypeScript first.

marshallswain committed:
Put ts first in hooks tabs

marshallswain committed:
Remove test tabs

daffl committed:
Add config and application changes

daffl committed:
Merge with WIP guide quick start docs

daffl committed:
Add deep outline to all API pages

marshallswain committed:
Padding for tabs without direct code blocks

marshallswain committed:
Split db table into core and ecosystem

marshallswain committed:
Add Logo and BlockQuote global components

marshallswain committed:
database adapter pages now show core modules

link to awesome-feathers for core modules

daffl committed:
Fix dead links and make localhost links codeblocks for now

daffl committed:
Some minor docs editing

marshallswain committed:
Clean up padding on tabs

8BallBomBom commented on docs/feathers-vs-loopback.md:
```diff
@@ -0,0 +1,17 @@
+# Feathers vs Sails
```
Should be Feathers vs LoopBack

8BallBomBom replied:
also keep up the good work 👍🏻

marshallswain replied:
Fixed.  Thanks!

marshallswain committed:
Begin db adapter pages

marshallswain committed:
update api/index

marshallswain committed:
Begin externalizing examples

marshallswain committed:
update icons/artwork

marshallswain committed:
fix typo Sails -> LoopBack

marshallswain committed:
Fix migration guide, add v3 to v4 page

marshallswain committed:
clean up migration guide

daffl committed:
Merge branch 'dove' of github.com:feathersjs/feathers into dove-docs

daffl committed:
Fix build

marshallswain committed:
custom line highlight color

marshallswain committed:
docs: update packages & update vitepress theme

daffl committed:
Initial work on new guide and other docs improvements

daffl committed:
Add chat migration

daffl committed:
Fix Knex migration dateformat

daffl committed:
Another v5 getting started guide update

daffl committed:
Further tweaks in the getting started guide

marshallswain committed:
allow customizing BlockQuote’s label

marshallswain committed:
Allow hiding label of language select

Also make the label bold.

marshallswain committed:
Fix Tab top margin

marshallswain committed:
Add simpler LanguageBlock component

marshallswain committed:
Add a global language select to the left sidebar

marshallswain committed:
Tabs —> LanguageBlock

marshallswain committed:
review starting.md

daffl committed:
More setup for the guides

daffl committed:
Merge with latest dove

daffl committed:
Merge branch 'dove' of github.com:feathersjs/feathers into dove-docs

marshallswain committed:
Update all packages

marshallswain committed:
fix grammar in language select part of guide

marshallswain committed:
remove navbar bg on home page in dark mode

marshallswain committed:
update components.d.ts

marshallswain committed:
update packages

marshallswain committed:
fix typo

marshallswain committed:
test.tpl.ts auth test cleans up its test user

daffl committed:
Merge branch 'dove-docs' of github.com:feathersjs/feathers into dove-docs

daffl committed:
Merge with latest dove

daffl committed:
Add compilation script for examples and update guide with latest generator file structure

daffl committed:
Adding some more in progress pages

daffl committed:
More docs for hooks, app, configuration and resolvers

mdartic committed:
Fix dispach > dispatch

Small typo

daffl committed:
More core API docs

marshallswain committed:
Merge pull request #2687 from mdartic/patch-1

Fix dispach > dispatch

daffl committed:
More API docs updated

daffl committed:
Improve layout and further docs

marshallswain committed:
clean up comment

marshallswain committed:
docs: update all dependencies

daffl committed:
Further API docs improvements

daffl committed:
Add inline markdown TypeScript to JavaScript transpiler

daffl committed:
Merge branch 'dove' of github.com:feathersjs/feathers into dove-docs

daffl committed:
Further API docs updates

blackmann committed:
chore(docs): Indicate how/when .parse of the strategy is used (#2712)

marshallswain committed:
updates to channels docs

marshallswain committed:
chore: update unocss deps

marshallswain committed:
docs: begin “troubleshooting” and “whats new” guides

marshallswain committed:
lint cleanup

marshallswain committed:
docs: split up schema intro

marshallswain committed:
docs: prettier cleanup for schema docs

marshallswain committed:
docs: grammar fix

marshallswain committed:
docs: more whats new in dove

marshallswain committed:
docs: make hooks vs resolvers an h3

daffl committed:
Merge with latest Dove

daffl committed:
Install actual CLI package

daffl committed:
Fix feathersjs/cli references

marshallswain committed:
docs: more what’s new in dove

daffl committed:
Add docs for custom Koa middleware

candicetate committed:
Spelling (#2716)

marshallswain committed:
docs: add “outdated” warning to top of CLI structure

marshallswain committed:
lint cleanup on api structure page

marshallswain committed:
update packages

marshallswain committed:
docs: finish documenting new features in dove

marshallswain committed:
update warning sections in whats new page

marshallswain committed:
update migrating guide

marshallswain committed:
Update expected release date.

marshallswain committed:
fix typo

marshallswain committed:
breaking: whitelist, allow renamed to `operators`

marshallswain committed:
docs: better wording for custom operators changes

marshallswain committed:
docs: collapse guides sidebar since it’s getting long

marshallswain committed:
docs: clean up what’s new section

marshallswain committed:
begin data loader docs

marshallswain committed:
fix links

marshallswain committed:
docs: upgrade to latest Vitepress

marshallswain committed:
docs: add dataloaders to what’s new

marshallswain committed:
docs: begin populating guide

marshallswain committed:
docs: note about mongodb data conversion

marshallswain committed:
docs: more details about schema

marshallswain committed:
docs: better docs for custom ajv and mongodb

marshallswain committed:
docs: add schema errors to troubleshooting guide

marshallswain committed:
fix dead link

marshallswain committed:
docs: clean up custom ajv examples in mongodb docs

marshallswain committed:
docs: switch mongodb resolver to be for data

instead of result

marshallswain committed:
docs: clean up wording in populating guide

marshallswain committed:
docs: add custom filters to migration guide

marshallswain committed:
docs: fix example

marshallswain committed:
docs: fix migration guide for filters

marshallswain committed:
docs: more operators cleanup

marshallswain committed:
Remove dataloader until it’s ready

candicetate committed:
Spelling edits (#2725)

Edits to spelling and phrasing

candicetate committed:
Spelling Updates (#2726)

candicetate committed:
Spelling edits (#2727)

Edits for spelling and phrasing.

candicetate committed:
Spelling Edits (#2728)

Edits to spelling and phrasing

candicetate committed:
Spelling Edits (#2729)

Edits to spelling and phrasing

candicetate committed:
Spelling Edits (#2730)

Edits for spelling and phrasing

candicetate committed:
Spelling Edits (#2731)

Edits to spelling and phrasing

candicetate committed:
Spelling Edits (#2732)

Edits to spelling and phrasing

candicetate committed:
Spelling Edits (#2733)

Edits to spelling and phrasing

daffl committed:
Merge with latest Dove

daffl committed:
Update docs dependencies and fix highlighting issue

daffl committed:
Finalize CSS based language selection

daffl committed:
Merge with latest dove

daffl committed:
docs: Fix language selection JS code blocks always being hidden

daffl committed:
docs: Fix line highlighting

marshallswain committed:
docs fix typo in custom ajv example

marshallswain committed:
docs add registering services to troubleshooting guide

marshallswain committed:
docs add ts-only content message

marshallswain committed:
docs fix client usage for custom service

marshallswain committed:
docs always import Infer as a type

marshallswain committed:
docs fix typos in rest.md

daffl committed:
Merge with latest dove

marshallswain committed:
docs: update to latest deps with new vitepress

marshallswain committed:
docs: fix errors during build

`document` was unavailable, so I tucked it inside the `onMounted` hook to allow the build to complete.

marshallswain committed:
docs: new overview page

This overview page focuses on architectural comparison and a brief history of Feathers.

daffl committed:
Update the guide for pre.30

marshallswain committed:
docs: update cli guide

marshallswain committed:
docs: updates to cli docs

anonimusprogramus committed:
chore(docs): Update events.md (#2787)

daffl committed:
Merge with latest dove

daffl committed:
Update getting started guide with pre.31 and add Github authentication guide

daffl committed:
Improve database API documentation

daffl committed:
Minor page edits

daffl commented:
Closing this PR since the new docs are available now at https://dove.feathersjs.com. We'll keep the branch around in case there is anything else we still need from it.
